hadoop-ozone/integration-test-s3/src/test/java/org/apache/hadoop/ozone/s3/awssdk/v2/AbstractS3SDKV2Tests.java

```diff
@@ -370,6 +377,215 @@ public void testCopyObject() {
     assertEquals("\"37b51d194a7513e45b56f6524f2d51f2\"", copyObjectResponse.copyObjectResult().eTag());
   }
 
+  @Test
+  public void testPutAndGetObjectTagging() {
+    final String bucketName = getBucketName();
+    final String keyName = getKeyName();
+    final String content = "test content";
+    s3Client.createBucket(b -> b.bucket(bucketName));
+
+    s3Client.putObject(b -> b
+            .bucket(bucketName)
+            .key(keyName),
+        RequestBody.fromString(content));
+
+    List<Tag> tags = Arrays.asList(
+        Tag.builder().key("env").value("test").build(),
+        Tag.builder().key("project").value("ozone").build()
+    );
+
+    s3Client.putObjectTagging(b -> b
+        .bucket(bucketName)
+        .key(keyName)
+        .tagging(Tagging.builder().tagSet(tags).build()));
+
+    GetObjectTaggingResponse getResponse = s3Client.getObjectTagging(b -> b
+        .bucket(bucketName)
+        .key(keyName));
+
+    assertEquals(tags.size(), getResponse.tagSet().size());
+    Map<String, String> tagMap = getResponse.tagSet().stream()
+        .collect(Collectors.toMap(Tag::key, Tag::value));
+    assertEquals("test", tagMap.get("env"));
+    assertEquals("ozone", tagMap.get("project"));
+  }
+
+  @Test
+  public void testDeleteObjectTagging() {
+    final String bucketName = getBucketName();
+    final String keyName = getKeyName();
+    final String content = "test content";
+    s3Client.createBucket(b -> b.bucket(bucketName));
+
+    s3Client.putObject(b -> b
+            .bucket(bucketName)
+            .key(keyName),
+        RequestBody.fromString(content));
+
+    List<Tag> tags = Arrays.asList(
+        Tag.builder().key("temp").value("data").build()
+    );
+
+    s3Client.putObjectTagging(b -> b
+        .bucket(bucketName)
+        .key(keyName)
+        .tagging(Tagging.builder().tagSet(tags).build()));
+
+    GetObjectTaggingResponse beforeDelete = s3Client.getObjectTagging(b -> b
+        .bucket(bucketName)
+        .key(keyName));
+    assertEquals(1, beforeDelete.tagSet().size());
+
+    s3Client.deleteObjectTagging(b -> b
+        .bucket(bucketName)
+        .key(keyName));
+
+    GetObjectTaggingResponse afterDelete = s3Client.getObjectTagging(b -> b
+        .bucket(bucketName)
+        .key(keyName));
+    assertTrue(afterDelete.tagSet().isEmpty());
+  }
+
+  @Test
+  public void testPutObjectTaggingExceedsLimit() {
+    final String bucketName = getBucketName();
+    final String keyName = getKeyName();
+    s3Client.createBucket(b -> b.bucket(bucketName));
+    s3Client.putObject(b -> b.bucket(bucketName).key(keyName),
+        RequestBody.fromString("content"));
+
+    List<Tag> tags = new ArrayList<>();
+    for (int i = 1; i <= 11; i++) {
```

Comment:
Could you please explain what `11` is?
We could use `TAG_NUM_LIMIT + 1` to make it clear.